be/src/exprs/vectorized/cast_expr_array.cpp

```diff
@@ -0,0 +1,202 @@
+    }
+    return cast_column;
+};
+
+// Cast string to array<ANY>
+ColumnPtr CastStringToArray::evaluate(ExprContext* context, vectorized::Chunk* input_chunk) {
+    ColumnPtr column = _children[0]->evaluate(context, input_chunk);
+    if (column->only_null()) {
+        return ColumnHelper::create_const_null_column(column->size());
+    }
+
+    PrimitiveType element_type = _cast_elements_expr->type().type;
+    ColumnViewer<TYPE_VARCHAR> src(column);
+    UInt32Column::Ptr offsets = UInt32Column::create();
+    NullColumn::Ptr null_column = NullColumn::create();
+
+    // 1. Split string with ',' delimiter
+    uint32_t offset = 0;
+    ColumnBuilder<TYPE_VARCHAR> slice_builder(src.size());
+    for (size_t i = 0; i < src.size(); i++) {
+        offsets->append(offset);
+        if (src.is_null(i)) {
+            null_column->append(1);
+            continue;
+        }
+        null_column->append(0);
+        Slice str = src.value(i);
+        if (str.starts_with("[")) {
+            str.data++;
+            str.size--;
+        }
+        if (str.ends_with("]")) {
+            str.size--;
+        }
+        if (!str.empty()) {
+            StringPiece str_piece(str.data, str.size);
+            std::vector<StringPiece> pieces;
+            SplitStringPieceToVector(str_piece, kArrayDelimeter, &pieces, false);
+
+            // Unquote slice for string type
+            if (element_type == TYPE_VARCHAR || element_type == TYPE_CHAR) {
+                for (auto& piece : pieces) {
+                    slice_builder.append(_unquote(Slice(piece.data(), piece.size())));
+                }
+            } else {
+                for (auto& piece : pieces) {
+                    slice_builder.append(Slice(piece.data(), piece.size()));
+                }
+            }
+            offset += pieces.size();
+        }
+    }
+    offsets->append(offset);
+
+    // 2. Cast string to specified type
+    ColumnPtr elements = slice_builder.build_nullable_column();
+    if (element_type != TYPE_VARCHAR && element_type != TYPE_CHAR) {
+        ChunkPtr chunk = std::make_shared<Chunk>();
+        SlotId slot_id = down_cast<ColumnRef*>(_cast_elements_expr->get_child(0))->slot_id();
+        chunk->append_column(elements, slot_id);
+        elements = ColumnHelper::cast_to_nullable_column(_cast_elements_expr->evaluate(context, chunk.get()));
+    }
+
+    // 3. Assemble elements into array column
+    ColumnPtr res = ArrayColumn::create(elements, offsets);
+    if (column->is_nullable()) {
+        res = NullableColumn::create(res, null_column);
+    }
+
+    return res;
+}
+
+Slice CastStringToArray::_unquote(Slice slice) {
+    while (slice.starts_with(" ")) {
+        slice.remove_prefix(1);
+    }
+    while (slice.ends_with(" ")) {
+        slice.remove_suffix(1);
+    }
+    if (slice.starts_with("\"") || slice.starts_with("'")) {
+        slice.remove_prefix(1);
+    }
+    if (slice.ends_with("\"") || slice.ends_with("'")) {
```

Comment:
- What's the result for `select cast('["1\']' as array<string>)`? Should it be `["\"1'"]` or `[null]`?
- And what's the result for `select cast('[a]' as array<string>)`? Should it be `["a"]` or `[null]`?

Reply:
The result should be changed into: 
1. `select cast('["1\']' as array<string>)` => `["\"1\'"]`, a string with escaped quote
2. `select cast('[a]' as array<string>)` => `["a"], which should be correct casted